src/qibolab/_core/instruments/emulator/utils.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+
+GHZ_TO_HZ = 1e9
+"""Converting GHz to Hz."""
+HZ_TO_GHZ = 1e-9
+"""Converting Hz to GHz."""
+
+
+def merge_results(a: dict, b: dict) -> dict:
+    """Merge dictionary together using np.column_stack."""
+    if len(a) == 0:
+        return b
+    if len(b) == 0:
+        return a
+    for key, value in b.items():
+        a[key] = np.column_stack((a[key], value))
+    return a
```

Comment:
```suggestion
    merged = a.copy()
    for key, value in b.items():
        x = (a[key],) if key in a else ()
        merged[key] = np.column_stack(x + (value,))
    return merged
```

This is the best I managed to make it more general, and avoid limit cases. However, inspecting the code requiring this function, it seemed to me that we can fully avoid this (simplifying and speeding up the whole process, working with arrays).

But at this point, it is easier if I just implement it in a commit, and propose a small PR for it.